### PR TITLE
Mockup Pastie Endpoints

### PR DESCRIPTION
**Pull request by carsontmoore**
First mock of Pastie endpoints on backend.  Addresses #15, #16, and #20.

### CODE REVIEW

File: server/apis/pasties-api.js [hunk 1]
```diff
@@ -10,3 +10,36 @@ PastiesAPI.get('/pasties/:id', function (req, res) {
...
+		.then(API.prep(200, res))
+		.catch( Pastie.NotFound, API.prep(404, res) )
+    .catch( API.catchUnexpectedErrors )
+})
+
+PastiesAPI.post('/pasties', function(req, res) {
+
+	Pastie.create(req.body)
+		.then(API.prep(201, res))
+		.catch( Pastie.InvalidFormat, API.prep(404, res) )
```
Comment (gilbert) [inline server/apis/pasties-api.js]: 404 means "Not found" – 400 more appropriately means "error caused by the client"

File: server/apis/pasties-api.js [hunk 2]
```diff
@@ -10,3 +10,36 @@ PastiesAPI.get('/pasties/:id', function (req, res) {
     .catch( Pastie.NotFound, API.prep(404, res) )
     .catch( API.catchUnexpectedErrors )
 })
+
+
+PastiesAPI.get('/pasties', function(req, res) {
+
+	Pastie.findWhere( req.query )
```
Comment (gilbert) [inline server/apis/pasties-api.js]: I think `Pastie.feedForUser( req.user.uid )` would be a more accurate mockup :)

File: server/apis/pasties-api.js [hunk 3]
```diff
@@ -10,3 +10,36 @@ PastiesAPI.get('/pasties/:id', function (req, res) {
...
+    .catch( API.catchUnexpectedErrors )
+})
+
+
+PastiesAPI.get('/pasties/public', function(req, res) {
+
+	Pastie.getPublic( req.query )
+		.then(API.prep(200, res))
+		.catch( Pastie.NotFound, API.prep(404, res) )
+    .catch( API.catchUnexpectedErrors )
```
Comment (gilbert) [inline server/apis/pasties-api.js:29]: Indenting

Comment (gilbert) [reply server/apis/pasties-api.js:29]: Also make sure you use spaces (2) instead of tabs

**Review by gilbert — CHANGES_REQUESTED** [review]
Nice job, this is pretty much what the API will end up looking like :)

File: server/apis/pasties-api.js (hunk 2 above)
Comment (shamsup) [reply server/apis/pasties-api.js]: That name makes more sense. 
Additionally, the `req.query` portion is for the potential of adding search further into the project, so would it be wise to still include that part? ie `Pastie.feedForUser(req.user.uid, req.query)`

Comment (gilbert) [reply server/apis/pasties-api.js]: Yes, that is perfectly ok :)

**Review by shamsup — APPROVED** [review]
Everything looks good

**Review by gilbert — APPROVED** [review]

**Pull request merged**